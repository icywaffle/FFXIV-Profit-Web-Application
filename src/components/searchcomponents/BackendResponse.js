import React from 'react'
import '../../../node_modules/uikit/dist/css/uikit.css'
import 'uikit'
function MainRecipe(baseinfo) {
    return (
        <section>
            <div className="uk-section uk-background-secondary uk-animation-slide-left uk-light">
                <div className="uk-container">
                    <div className="uk-width-4-5 uk-float-right">
                        <h2>Analysis</h2>
                        <h4>
                            <img
                                className="itemicon"
                                alt="icon"
                                src={"/icon/0" + baseinfo.Recipes.IconID.toString().substr(0, 2) + "000/0" + baseinfo.Recipes.IconID.toString() + ".png"}
                            />
                            {baseinfo.Recipes.Name}
                        </h4>
                        <ul className="uk-list">
                            <li>On Market: {baseinfo.Prices.OnMarketboard.toString()}</li>
                            <li>Marketboard Price: <span className=" pricenumber">{baseinfo.Prices.LowestMarketPrice}</span>
                            </li>
                            <li>
                                Material Costs: <span className="pricenumber">{baseinfo.Profits.MaterialCosts}</span>
                            </li>
                            <li>
                                Profits: <span className="pricenumber">{baseinfo.Profits.Profits}</span>
                            </li>
                            <li>Profit Percentage: {baseinfo.Profits.ProfitPercentage}%</li>
                            <li className="datetime" value="{baseinfo.Profits.Added}"> </li>
                        </ul>
                    </div>
                </div>
            </div>
        </section>
    )
}

function MainMaterials(baseinfo, matinfo) {
    const materialList = baseinfo.Recipes.IngredientNames.map((material, index) => {
        // If there is no material, just return nothing
        if (material === "") {
            return ""
        }

        // For each material, we want to make it an accordion content.
        // So we would need to place all the information here
        const ingredientID = baseinfo.Recipes.IngredientID[index]

        const marketInfo = (matinfo.Prices[ingredientID] === undefined)
            ? ""
            : (<ul className="uk-list">
                <li>
                    Lowest Market Price :
                    <span className="pricenumber">{matinfo.Prices[ingredientID].LowestMarketPrice}</span>
                </li>
                <li>
                    On Marketboard : {matinfo.Prices[ingredientID].OnMarketboard}
                </li>
                <li>
                    Added : <span className="datetime">{matinfo.Prices[ingredientID].Added}</span>
                </li>
            </ul>
            )

        return (
            <li>
                <a className="uk-accordion-title" href="">
                    <img
                        className="itemicon"
                        src={"/icon/0" + baseinfo.Recipes.IngredientIconID[index].toString().substr(0, 2) + "000/0" + baseinfo.Recipes.IngredientIconID[index].toString() + ".png"}
                    />
                    <span>{material}</span>
                    <span className="uk-text-meta">
                        x{baseinfo.Recipes.IngredientAmounts[index]}</span>

                </a>
                <div className="uk-accordion-content" aria-hidden="true" hidden>
                    {marketInfo}
                </div>
            </li>
        )
    })
    return (
        <section>
            <div className="uk-section uk-background-secondary uk-animation-slide-left uk-light">
                <div className="uk-container">
                    <div className="uk-width-4-5 uk-float-right">
                        <h2>Materials</h2>
                        <ul className="uk-list" uk-accordion="multiple: true">
                            {materialList}
                        </ul>
                    </div>
                </div>
            </div>
        </section>
    )
}
function BackendResponse(props) {
    const mainrecipe = MainRecipe(props.recipeData.MainRecipe)
    const mainmaterials = MainMaterials(props.recipeData.MainRecipe, props.recipeData.InnerRecipes)
    return (<div>
        {mainrecipe}
        {mainmaterials}
    </div>)
}

export default BackendResponse