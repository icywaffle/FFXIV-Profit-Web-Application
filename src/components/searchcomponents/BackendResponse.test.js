import React from "react"
import { shallow, mount } from "enzyme"

// Basic Mock Recipe Data that would be returned from the database
const mockRecipeData = {
    "MainRecipe": {
        "Recipes": {
            "Name": "Seeing Horde Axe",
            "IconID": 31518,
            "ItemResultTargetID": 24322,
            "ID": 33180,
            "CraftTypeTargetID": 1,
            "RecipeLevelTable": {
                "ClassJobLevel": 60,
                "Stars": 3
            },
            "AmountResult": 1,
            "IngredientID": [
                14146,
            ],
            "IngredientIconID": [
                20824,
            ],
            "IngredientNames": [
                "High Mythrite Ingot",
            ],
            "IngredientAmounts": [
                3,
            ],
            "IngredientRecipes": [
                [
                    31482,
                    31483
                ],
            ],
            "Added": 1563260884
        },
        "Prices": {
            "ItemID": 24322,
            "HistoryPrice": 0,
            "LowestMarketPrice": 0,
            "OnMarketboard": false,
            "Added": 1569623257
        },
        "Profits": {
            "RecipeID": 33180,
            "ItemID": 24322,
            "MaterialCosts": 1418852,
            "Profits": 471147,
            "ProfitPercentage": 33.21,
            "Added": 1563265228,
            "Name": "Seeing Horde Axe",
            "IconID": 31518,
            "CraftTypeTargetID": 1,
            "RecipeLevelTable": {
                "ClassJobLevel": 60,
                "Stars": 3
            }
        }
    },
    InnerRecipes: {
        Recipes: {
            "31482": {
                Name: "High Mythrite Ingot",
                IconID: 20824,
                ItemResultTargetID: 14146,
            }
        },
        Prices: {
            "14146": {
                ItemID: 14146,
                HistoryPrice: 0,
                LowestMarketPrice: 0,
                OnMarketboard: false,
                Added: 1569623257
            }
        },
        Profits: {
            "31482": {
                RecipeID: 31482,
                ItemID: 14146,
                MaterialCosts: 10315,
                Profits: 9685,
                ProfitPercentage: 93.89,
                Added: 1563265224,
                Name: "High Mythrite Ingot",
                IconID: 20824,
                CraftTypeTargetID: 1,
                RecipeLevelTable: {
                    ClassJobLevel: 60,
                    Stars: 3
                }
            },
        }
    }
}

import BackendResponse from "./BackendResponse"
describe("BackendResponse", () => {
    it("should render when given specific database information", () => {

        const component = shallow(<BackendResponse recipeData={mockRecipeData} />)

        expect(component).toMatchSnapshot()
    })
    it("should show Database Down! if no data was given", () => {
        const component = mount(<BackendResponse recipeData={""} />)
        expect(component.text()).toBe("Database may be down!")
    })
})