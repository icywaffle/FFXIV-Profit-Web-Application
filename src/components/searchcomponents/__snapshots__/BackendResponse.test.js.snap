// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendResponse should render when given specific database information 1`] = `
<div>
  <section>
    <div
      className="uk-section uk-animation-slide-left"
    >
      <div
        className="uk-container"
      >
        <div
          className="uk-width-4-5 uk-float-right"
        >
          <h2>
            Analysis
          </h2>
          <h4>
            <img
              alt="icon"
              className="itemicon"
              src="/icon/031000/031518.png"
            />
            Seeing Horde Axe
          </h4>
          <ul
            className="uk-list"
          >
            <li>
              On Market: 
              false
            </li>
            <li>
              Marketboard Price: 
              <span
                className=" pricenumber"
              >
                0
              </span>
            </li>
            <li>
              Material Costs: 
              <span
                className="pricenumber"
              >
                1418852
              </span>
            </li>
            <li>
              Profits: 
              <span
                className="pricenumber"
              >
                471147
              </span>
            </li>
            <li>
              Profit Percentage: 
              33.21
              %
            </li>
            <li
              className="datetime"
              value="{baseinfo.Profits.Added}"
            >
               
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
  <section>
    <div
      className="uk-section uk-background-secondary uk-animation-slide-left uk-light"
    >
      <div
        className="uk-container"
      >
        <div
          className="uk-width-4-5 uk-float-right"
        >
          <h2>
            Materials
          </h2>
          <ul
            className="uk-list"
            uk-accordion="multiple: true"
          >
            <li>
              <a
                className="uk-accordion-title"
                href=""
              >
                <img
                  className="itemicon"
                  src="/icon/020000/020824.png"
                />
                <span>
                  High Mythrite Ingot
                   
                </span>
                <span
                  className="uk-text-meta"
                >
                  x
                  3
                </span>
              </a>
              <div
                className="uk-accordion-content"
              >
                <ul
                  className="uk-list"
                >
                  <li>
                    Lowest Market Price :
                    <span
                      className="pricenumber"
                    >
                      0
                    </span>
                  </li>
                  <li>
                    On Marketboard : 
                  </li>
                  <li>
                    Added : 
                    <span
                      className="datetime"
                    >
                      1569623257
                    </span>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</div>
`;
